### PR TITLE
api: New internal endpoint for getting user info

### PR DESCRIPTION
**Pull request by alfrunes**
Updated endpoint `POST /api/(management|internal)/v1/useradm/tenants/:id/users` with an `id` parameter to filter by user `id`s (needed by MEN-3749/3750).

### CODE REVIEW

**Comment by alfrunes** [comment]
Not ready for review just yet, need to change the API a tiny bit...

**Review by tranchitella — COMMENTED** [review]
To be honest, I would have implemented it as a GET with multiple id params:
`/api/internal/v1/useradm/tenants/:id/users?id=1&id=2&id=3`

Anyway, it is not an important thing, it is an internal end-point, but using "POST" to get data is a bit counterintuitive.

**Comment by alfrunes** [comment]
> To be honest, I would have implemented it as a GET with multiple id params:
> `/api/internal/v1/useradm/tenants/:id/users?id=1&id=2&id=3`
> 
> Anyway, it is not an important thing, it is an internal end-point, but using "POST" to get data is a bit counterintuitive.

TBH, I agree. My only concern is if there's a limit on the length on the start-line of an HTTP request that it could not return an error (414) if the URL is too long.

**Comment by tranchitella** [comment]
@alfrunes that's a good point, and I do not know the specific scenario where we consume this end-point, if this can be a problem let's stick to the POST method.

**Comment by alfrunes** [comment]
@tranchitella I see there is a default limit of 1MiB for request header in golang http. With the current restrictions in the auditlogs service this will be a very tight corner-case where this can actually fail and should practically never happen.
I'll change the endpoint to `GET /users` instead and add query parameters for filter restrictions as you suggested :+1:

**Comment by tranchitella** [comment]
@alfrunes thanks for taking my suggestion into account.

**Comment by alfrunes** [comment]
@tranchitella I just updated the management EP as well while I was at it :)

File: api/http/api_useradm.go [hunk 1]
```diff
@@ -91,8 +92,6 @@ func (i *UserAdmApiHandlers) GetApp() (rest.App, error) {
 		rest.Get(uriManagementSettings, i.GetSettingsHandler),
 	}
 
-	routes = append(routes)
-
```
Comment (tranchitella) [inline api/http/api_useradm.go:95]: Nice clean-up, these lines are there since 2016!

File: docs/internal_api.yml [hunk 2]
```diff
@@ -173,6 +173,75 @@ paths:
...
+          in: query
+          type: integer
+          description: >
+            Filter users created before timestamp (UNIX timestamp).
+          required: false
+        - name: updated_after
+          in: query
+          type: integer
+          description: >
+            Filter users created after timestamp (UNIX timestamp).
```
Comment (tranchitella) [inline docs/internal_api.yml:219]: *updated*

File: docs/internal_api.yml [hunk 3]
```diff
@@ -173,6 +173,75 @@ paths:
...
+          required: false
+        - name: updated_before
+          in: query
+          type: integer
+          description: >
+            Filter users created before timestamp (UNIX timestamp).
```
Comment (tranchitella) [inline docs/internal_api.yml:225]: *updated*

File: docs/management_api.yml [hunk 4]
```diff
@@ -78,6 +78,45 @@ paths:
...
+          in: query
+          type: integer
+          description: >
+            Filter users created before timestamp (UNIX timestamp).
+          required: false
+        - name: updated_after
+          in: query
+          type: integer
+          description: >
+            Filter users created after timestamp (UNIX timestamp).
```
Comment (tranchitella) [inline docs/management_api.yml:112]: *updated*

File: docs/management_api.yml [hunk 5]
```diff
@@ -78,6 +78,45 @@ paths:
...
+          required: false
+        - name: updated_before
+          in: query
+          type: integer
+          description: >
+            Filter users created before timestamp (UNIX timestamp).
```
Comment (tranchitella) [inline docs/management_api.yml:118]: *updated*

File: model/user.go [hunk 6]
```diff
@@ -150,3 +152,59 @@ func checkPwd(password string) error {
...
+	Email []string `json:"email,omitempty"`
+
+	CreatedAfter  *time.Time `json:"created_after,omitempty"`
+	CreatedBefore *time.Time `json:"created_before,omitempty"`
+
+	UpdatedAfter  *time.Time `json:"updated_after,omitempty"`
+	UpdatedBefore *time.Time `json:"updated_before,omitempty"`
+}
+
+func (fltr *UserFilter) ParseForm(form url.Values) error {
```
Comment (tranchitella) [inline model/user.go:167]: It would be straightforward to implement user test.go and cover these methods...

**Review by tranchitella — APPROVED** [review]
A few minor comments, pre-approving!

File: model/user.go (hunk 6 above)
Comment (alfrunes) [reply model/user.go:167]: Right! I totally forgot about these methods!

**Pull request merged**